app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -125,6 +142,10 @@ public void onReceive(Context ctx, Intent intent) {
         updateCurrentBgInfo();
     }
 
+    public static Context getContext() {
```

Comment:
As I said, I cannot use addToQueue to get the context and initialise pebbleSync. It keeps creating PebbleSync s. 

 
John Stevens
M - +61 418526349

> On 20 May 2015, at 07:33, Stephen Black <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Home.java:
> 
> > @@ -125,6 +142,10 @@ public void onReceive(Context ctx, Intent intent) {
> >          updateCurrentBgInfo();
> >      }
> >  
> > +    public static Context getContext() {
> the service that calls into a class (or the activity) would pass its own context along, it seems the only place you are using this is in the BgSendQueue class which is handed a context whenever it is used already, we should just use that same context
> 
> —
> Reply to this email directly or view it on GitHub.
>